From 0190d88675ed8d982bfd1aa670c07f79f7af9a59 Mon Sep 17 00:00:00 2001
Message-Id: <0190d88675ed8d982bfd1aa670c07f79f7af9a59.1421835210.git.chang-joon.lee@intel.com>
In-Reply-To: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
References: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Mon, 8 Dec 2014 13:21:13 +0000
Subject: [PATCH 36/63] REVERTME [VPG]: drm/i915: Remove mmio_flip::ring_id

The display code in DRM nightly is significantly different to that in
GMin. One important difference with respect to the following seqno ->
request patches is the removal of the 'ring_id' field from the MMIO
flip structure. This patch does that removal so that the following
patches can more closely follow their source in the official DRM tree.

Once the display code has been updated to a newer version by the next
forklift, this patch can be dropped.

Change-Id: Ie8ff1e35a34fff5366591df9b1afcf99fde3ba21
For: VIZ-4377
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    3 +--
 drivers/gpu/drm/i915/intel_drv.h     |    1 -
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 22027d2..9edad00 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -10437,7 +10437,7 @@ void intel_notify_mmio_flip(struct intel_engine_cs *ring)
 		if (mmio_flip->req == NULL)
 			continue;
 
-		if (ring->id != mmio_flip->ring_id)
+		if (i915_gem_request_get_ring(mmio_flip->req) != ring)
 			continue;
 
 		if (i915_seqno_passed(seqno, i915_gem_request_get_seqno(mmio_flip->req))) {
@@ -10476,7 +10476,6 @@ static int intel_queue_mmio_flip(struct drm_device *dev,
 	spin_lock_irqsave(&dev_priv->mmio_flip_lock, irq_flags);
 	i915_gem_request_assign(&intel_crtc->mmio_flip.req,
 				obj->last_write_req);
-	intel_crtc->mmio_flip.ring_id = obj->ring->id;
 	spin_unlock_irqrestore(&dev_priv->mmio_flip_lock, irq_flags);
 
 	/*
diff --git a/drivers/gpu/drm/i915/intel_drv.h b/drivers/gpu/drm/i915/intel_drv.h
index c4af579..9f9e8af 100644
--- a/drivers/gpu/drm/i915/intel_drv.h
+++ b/drivers/gpu/drm/i915/intel_drv.h
@@ -384,7 +384,6 @@ struct intel_pipe_wm {
 
 struct intel_mmio_flip {
 	struct drm_i915_gem_request *req;
-	u32 ring_id;
 };
 
 struct intel_disp_reg {
-- 
1.7.9.5

